python/tk_multi_publish2/base_hooks/collector.py

```diff
@@ -155,10 +166,224 @@ def update_value(self):
                 # Only overwrite valid values
                 item.properties[self._name] = self._value
 
+            # Emit that our value has changed, after updating the values
+            # otherwise other widgets still get the old properties value
+            self.value_changed.emit()
+
         @property
         def items(self):
             return self._items
 
+    class FieldsPropertyWidget(PluginBase.ValueWidgetBaseClass):
+        """
+        A widget class representing a template setting.
+        """
+
+        # Signal for when a setting field has changed
+        field_changed = QtCore.Signal()
+
+        @property
+        def items(self):
+            return self._items
+
+        def __init__(self, parent, hook, items, name, **kwargs):
+
+            super(CollectorPlugin.FieldsPropertyWidget, self).__init__(
+                parent, hook, name, value=None, **kwargs)
+
+            self._fields = {}
+            self._field_widgets = {}
+            self._layout = QtGui.QVBoxLayout(self)
+            self._items = items
+
+            # list of non editable fields for the widget
+            self._non_editable_fields = kwargs.pop("non_editable_fields", list())
```

Comment:
The default should be non-editable

Reply:
Wild-card matching seems fine to me.

Reply:
`self._non_editable_fields` now does an fnmatch on the contents of the list

Reply:
This still needs to be addressed. The default should be non-editable, so this property should be "editable_fields".